pkg/internal/apply/layerApplier.go

```diff
@@ -179,35 +226,44 @@ func (a KubectlLayerApplier) decodeList(layer layers.Layer,
 	return hrs, nil, nil
 }
 
-// Apply an AddonLayer to the cluster.
-func (a KubectlLayerApplier) Apply(layer layers.Layer) (err error) {
-	sourceDir := layer.GetSourcePath()
-	a.logInfo("Applying AddonsLayer", layer)
+func (a KubectlLayerApplier) checkSourcePath(layer layers.Layer) (sourceDir string, err error) {
+	sourceDir = layer.GetSourcePath()
 	info, err := os.Stat(sourceDir)
 	if os.IsNotExist(err) {
 		a.logDebug("source directory not found", layer)
-		return fmt.Errorf("source directory (%s) not found for AddonsLayer %s/%s",
+		return sourceDir, fmt.Errorf("source directory (%s) not found for AddonsLayer %s/%s",
 			sourceDir, "", layer.GetName())
 	}
 	if os.IsPermission(err) {
 		a.logDebug("source directory read permission denied", layer)
-		return fmt.Errorf("read permission denied to source directory (%s) for AddonsLayer %s/%s",
+		return sourceDir, fmt.Errorf("read permission denied to source directory (%s) for AddonsLayer %s/%s",
 			sourceDir, "", layer.GetName())
 	}
 	if err != nil {
 		a.logError(err, "error while checking source directory", layer)
-		return fmt.Errorf("error while checking source directory (%s) for AddonsLayer %s/%s",
+		return sourceDir, fmt.Errorf("error while checking source directory (%s) for AddonsLayer %s/%s",
 			sourceDir, "", layer.GetName())
 	}
 	if !info.IsDir() {
 		// I'm not sure if this is an error, but I thought I should detect and log it
 		a.logInfo("source path is not a directory", layer)
 	}
+	return sourceDir, nil
+}
+
+// Apply an AddonLayer to the cluster.
+func (a KubectlLayerApplier) Apply(layer layers.Layer) (err error) {
+	a.logInfo("Applying AddonsLayer", layer)
+
+	sourceDir, err := a.checkSourcePath(layer)
+	if err != nil {
+		return err
+	}
 
-	output, err := a.kubectl.Apply(sourceDir).WithLogger(layer.GetLogger()).Run()
-	//output, err := a.kubectl.Apply(sourceDir).WithLogger(layer.GetLogger()).DryRun()
+	//output, err := a.kubectl.Apply(sourceDir).WithLogger(layer.GetLogger()).Run()
+	output, err := a.kubectl.Apply(sourceDir).WithLogger(layer.GetLogger()).DryRun()
```

Comment:
Really not sure of `GetLogger()` being on the layer interface

Reply:
Yeah, we're planning to move that in a bit.